src/Application/Items/Commands/UpgradeUserItemCommand.cs

```diff
@@ -0,0 +1,97 @@
+using AutoMapper;
+using Crpg.Application.Common;
+using Crpg.Application.Common.Interfaces;
+using Crpg.Application.Common.Mediator;
+using Crpg.Application.Common.Results;
+using Crpg.Application.Common.Services;
+using Crpg.Application.Items.Models;
+using Crpg.Domain.Entities.Items;
+using Crpg.Domain.Entities.Users;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using NetTopologySuite.Index.HPRtree;
+using LoggerFactory = Crpg.Logging.LoggerFactory;
+
+namespace Crpg.Application.Items.Commands;
+
+public record UpgradeUserItemCommand : IMediatorRequest<UserItemViewModel>
+{
+    public int UserItemId { get; init; }
+    public int UserId { get; init; }
+
+    internal class Handler : IMediatorRequestHandler<UpgradeUserItemCommand, UserItemViewModel>
+    {
+        private static readonly ILogger Logger = LoggerFactory.CreateLogger<UpgradeUserItemCommand>();
+
+        private readonly ICrpgDbContext _db;
+        private readonly IMapper _mapper;
+        private readonly IActivityLogService _activityLogService;
+
+        public Handler(ICrpgDbContext db, IMapper mapper, IActivityLogService activityLogService)
+        {
+            _db = db;
+            _mapper = mapper;
+            _activityLogService = activityLogService;
+        }
+
+        public async Task<Result<UserItemViewModel>> Handle(UpgradeUserItemCommand req, CancellationToken cancellationToken)
+        {
+            var user = await _db.Users
+                .Include(u => u.Items)
+                .FirstOrDefaultAsync(u => u.Id == req.UserId, cancellationToken);
+
+            var userItems = await _db.UserItems
+                .Where(ui => user!.Items.Contains(ui))
+                .Include(ui => ui.Item)
+                .ToListAsync();
+
```

Comment:
I'm still confused about what you're trying to do here. It seems that you are loading a second time the user items here. You might want to follow a ef core tutorial https://learn.microsoft.com/en-us/aspnet/core/data/ef-mvc/intro

Reply:
I just did not now about ThenInclude()